COMET/misc_plugins/PlotScripts/analysis_scripts/IV_PQC.py

```diff
@@ -45,38 +66,760 @@ def __init__(self, data, configs):
+
+            if "1C2" not in data[samplekey]["data"]:
+
+                self.log.warning("Full depletion calculation could not be done for data set: {}".format(samplekey))
+
+            else:
+
+                self.log.debug("Data: {}".format(samplekey))
+
+                sample = deepcopy(data[samplekey])
+
+                df = pd.DataFrame({"xaxis": sample["data"]["Voltage"], "yaxis": sample["data"]["1C2"]})
+
+                df = df.dropna()
+
+                # Loop one time from the right side and from the left, to get both slopes
+
+                LR2 = 0 # r^2 values for both sides
+
+
+                for idx in range(5, len(df)-5):
+
+                    # Left
+
+                    slope_left, intercept_left, r_left, p_value, std_err_left = linregress(df["xaxis"][:-idx],df["yaxis"][:-idx])
+
+                    r2_left = r_left * r_left
+
+                    self.log.debug("Left side fit: Slope {}, intercept: {}, r^2: {}, std: {}".format(
+
+                        slope_left, intercept_left, r2_left, std_err_left)
+
+                    )
+
+
+                    # See if the r2 value has increased and store end points
+
+                    if r2_left >= LR2:
+
+                        LR2 = r2_left
+
+                        LeftEndPoints = (
+
+                            (df["xaxis"][0], intercept_left),
+
+                            (df["xaxis"][idx], slope_left * df["xaxis"][idx] + intercept_left)
+
+                        )
+
+                        Left_stats[i] = (LeftEndPoints, slope_left, intercept_left, r_left, p_value, std_err_left)
+
+
+
+        #Find the right fit by averaging on the final 20 points
+        average_right = np.mean(list(df['yaxis'][-20:]))
+        RightEndPoints =[(df['xaxis'][len(df['xaxis'])-20],average_right),(df['xaxis'][len(df['xaxis'])-1],average_right)]
+
+        # Find the line intersection
+        full_depletion_voltages = line_intersection(LeftEndPoints, RightEndPoints)
+
+
+        # Add vertical line for full depletion
+
+
+
+        vline = hv.VLine(full_depletion_voltages[0]).opts(color='black', line_width=5.0)
+
+
+        # Add slopes
+
+        left_line = np.array([[0, np.median(Left_stats[:,2])],[full_depletion_voltages[0], full_depletion_voltages[1]]])
+
+        left_line = hv.Curve(left_line).opts(color='grey')
+
+        right_line = hv.HLine(average_right).opts(color='grey')
+
+        # Add text
+
+        self.log.info('Full depletion voltage: {} V'
+
+                        .format(np.round(full_depletion_voltages[0], 2)))
+
+
+        text = hv.Text(230, (5e+21), 'Depletion voltage: {} V'.format(np.round(full_depletion_voltages[0], 2))).opts(style=dict(text_font_size='20pt'))
+
+
+        # Update the plot specific options if need be
+
+        returnPlot = plot * vline * right_line * left_line * text
+
+        returnPlot = customize_plot(returnPlot, "1C2", configs["IV_PQC"], **addConfigs)
+
+        return returnPlot, full_depletion_voltages[0]
```

Comment:
Please restructure the code and remove unecessary newlines. Block together what needs to be blocked. But only use newlines if necessary. I have a really hard time reading it. Furthermore, use subfunctions to get a more streamlined run function. I cannot see where one analysis starts and another ends. Thanks.

Furthermore, use my comments from the other files etc and apply it to this one as well. Then re-submit =)